yt/frontends/flash/data_structures.py

```diff
@@ -53,6 +53,15 @@ def _detect_output_fields(self):
             ("flash", s.decode("ascii", "ignore"))
             for s in self._handle["/unknown names"][:].flat
         ]
+        # add custom/scratch fields:
+        srp = self._handle["/string runtime parameters"]
+        for i in range(srp.shape[0]):
+            k = srp[i][0]
+            if k.startswith(b"plot_grid_var"):
+                v = srp[i][1].decode("ascii", "ignore")[:4]
+                if v != "none":
+                    self.field_list.append(("flash", v))
```

Comment:
```suggestion
        for k, v in srp:
            fname = v.decode("ascii", "ignore")[:4]
            if k.startswith(b"plot_grid_var") and fname != "none":
                self.field_list.append(("flash", fname))
```

Reply:
This looks like an improvement over the original *as long as* we are sure that iterating over `srp` in the way you propose still works with the way `string runtime parameters` are stored.

Reply:
of course ! I'm taking a bet here :)

Reply:
@cbyrohl, what's your thinking on this ? :-)